src/phone_number/mobile.rs

```diff
@@ -5,7 +5,8 @@ lazy_static! {
     static ref MOBILE_NUMBER_REGEX: Regex =
         Regex::new(r"(\+98|0|98|0098)?(9\d{2})+(\d{3})+(\d{4})$").unwrap();
```

Comment:
This regex is so ugly! It matches a lot of wrong stuff! like `9339339339339339339339911911911911911911`, let me see if we can have something better. Even the `$` at the end looks useless to me :-/

Reply:
At least all of the existing tests pass with this: `^(\+98|0|98|0098)(9\d{2})(\d{3})(\d{4})$`
I would suggest adding `9339339339339339339339911911911911911911` as a test case also :D
And maybe more test cases, for example, I'm not sure if it supposed to work with `9123456789` or not!